examples/new_member.py

```diff
@@ -9,7 +9,7 @@ class MyClient(discord.Client):
 
     async def on_member_join(self, member):
         guild = member.guild
-        await guild.system_channel.send('Welcome {0.mention} to {1.name}!'.format(member, guild))
+        await guild.channels[1].send('Welcome {0.mention} to {1.name}!'.format(member, guild))
```

Comment:
It may return an IndexError, moreover it can also be a channel categorie or a voice channel.
You should check if there's an available TextChannel before, with something like 
`if sum(1 for channel in guild.channels if type(channel) == discord.TextChannel):`
And maybe check if the bot can send message
```
if sum(1 for channel in guild.channels if type(channel) == discord.TextChannel):
    for channel in guild.channels:
        if type(channel) == discord.TextChannel:
            if channel.permissions_for(guild.me).send_messages:
                return await channel.send('Welcome {0.mention} to {1.name}!'.format(member, guild))
```